### DISCUSSION TITLE
Use exported enums in Vue

### DISCUSSION THREAD

**Discussion by nckrtl** [spatie/laravel-data#825] [Q&A]
I'm trying to using the enums exported by this package in Vue, but they don't seem usable. I'm exporting my enums to native TS enums with the config option: `'transform_to_native_enums' => true,`. The generated enums in generated.d.ts are looking fine: 

```
declare namespace App.Enums {
export enum DataResourceType { 'SERVER' = 'server', 'WEBSITE' = 'website' };
}
```

However I can't reference the enum as I get an "Can't find variable: App" error. As I understand the enums are only available for Typescript references but can't be used as values in Vue. Is there a way to accomplish this? Ideally I'd like to use the enums in generated.d.ts instead of manually maintaining an separate enums file.

**Comment by rubenvanassche** [comment]
Hi @nckrtl,

We (at Spatie) actually don't use native enums (so the value is `false`) for the flag meaning the enum will be exported as a union of string: 'server' | 'website'. So you can just use strings in TS and still have support for type checking.

**Reply by nckrtl** [reply]
That makes sense, was hoping I was missing something to use enums in JS/TS as well. Strings it will be then. Thanks Ruben.

**Comment by aphofstede** [comment]
String enums are very useful for using actual values in FE code. I think it would be a great addition for those cases where you consciously opt into string enum use.
